lib/Conversion/Utils/Utils.cpp

```diff
@@ -166,6 +166,33 @@ castIndexVectorToInt64Vector(OpBuilder &b, Location loc,
   return intValues;
 }
 
+SmallVector<Value> getDiagEmbedResultShape(OpBuilder &b, Location loc, Value tensor, int64_t  offset, int64_t  dim1, int64_t  dim2) {
+  auto inputType = tensor.getType().cast<RankedTensorType>();
+  auto inputRank = inputType.getRank();
+  auto resultRank = inputRank + 1;
+  
+  SmallVector<Value> resultShape;
+  Value constZero = b.create<arith::ConstantIndexOp>(loc, 0);
+  Value constNegOne = b.create<arith::ConstantIndexOp>(loc, -1);
+  Value constOffset = b.create<arith::ConstantIndexOp>(loc, offset);
+  Value isNegOffset = b.create<arith::CmpIOp>(loc, arith::CmpIPredicate::slt, constOffset, constZero);
+  Value mulOffsetNegOne = b.create<arith::MulIOp>(loc, constOffset, constNegOne);
+  Value absOffset = b.create<arith::SelectOp>(loc, isNegOffset, mulOffsetNegOne, constOffset);
```

Comment:
nit: I think you can use `math::AbsIOp`

Reply:
Good suggestion! I'm not very well versed in all available ops and dilects yet. Replaced with `math::AbsIOp`